lib/sqlalchemy/ext/declarative/api.py

```diff
@@ -229,8 +229,7 @@ class HasIdMixin(object):
                 @declared_attr.cascading
                 def id(cls):
                     if has_inherited_table(cls):
-                        return Column(
-                            ForeignKey('myclass.id'), primary_key=True)
+                        return Column('some_id', ForeignKey('myclass.id'))
```

Comment:
**Actual behavior** of previous version is:  `raise exc.ArgumentError`
```bash
File "sqlalchemy/ext/declarative/base.py", line 609, in _setup_inheritance
    raise exc.ArgumentError(
sqlalchemy.exc.ArgumentError: Column 'id' on class <class '__main__.MySubClass'> conflicts with existing column 'myclass.id'
```
This example was broken in [commit](https://github.com/sqlalchemy/sqlalchemy/commit/dce32062236001ed89258cfe60c89bbca5db18c5) 

**The reason is**: Example was changed by analogy with another (see [Mixing in Columns in Inheritance Scenarios](https://github.com/sqlalchemy/sqlalchemy/blob/master/doc/build/orm/extensions/declarative/mixins.rst)) (but in `Mixins ...` it was example with "Joined Table Inheritance"). However, in this example it is "Single Table Inheritance", and as a result, it is impossible to apply the column “id” twice.

Addition see for description in the next paragraph under example in documentation. Quote: 
> ``MyClass`` underneath the attribute named ``some_id``

My version create next sql:
```sql
CREATE TABLE myclass (
	id INTEGER NOT NULL, 
	some_id INTEGER, 
	PRIMARY KEY (id), 
	FOREIGN KEY(some_id) REFERENCES myclass (id)
)
```